src/convert/metadataConverter.ts

```diff
@@ -63,20 +63,22 @@ export class MetadataConverter {
             if (destructiveChangesTypes.length) {
               // for each of the destructive changes in the component set, convert and write the correct metadata
               // to each manifest
-              destructiveChangesTypes.map((destructiveChangesType) => {
-                const file = this.getDestructiveManifest(destructiveChangesType);
-                const destructiveManifestContents = cs.getPackageXml(4, destructiveChangesType);
-                const destructiveManifestPath = join(packagePath, file);
-                tasks.push(promises.writeFile(destructiveManifestPath, destructiveManifestContents));
-              });
+              tasks.push(
+                destructiveChangesTypes.map(async (destructiveChangesType) =>
+                  promises.writeFile(
+                    join(packagePath, this.getDestructiveManifest(destructiveChangesType)),
+                    await cs.getPackageXml(4, destructiveChangesType)
+                  )
+                )
+              );
```

Comment:
This doesn't seem right. Won't this cause `tasks` to end up with a nested array of `promises.writeFile` return types?
e.g. `tasks = [task1, task2, [promises.writeFile1, promises.writeFile2, promises.writeFile3], task3, task4, ...]`

Why are we even using `map` if the intent is just to iterate through the `destructiveChangesTypes` do some stuff and add them to `tasks`? Wouldn't `.forEach` be better and much clearer in intent?
```suggestion
              destructiveChangesTypes.forEach(async (destructiveChangesType) =>
                tasks.push(
                  promises.writeFile(
                    join(packagePath, this.getDestructiveManifest(destructiveChangesType)),
                    await cs.getPackageXml(4, destructiveChangesType)
                  )
                )
              );
```

Reply:
`tasks` array is now typed so that mistake won't happen

Reply:
You're right about `.forEach` not working with `async` down below so how about a `for ... of` loop instead?

Example:
```suggestion
              for (const destructiveChangesType of destructiveChangesTypes) {
                tasks.push(
                  promises.writeFile(
                    join(packagePath, this.getDestructiveManifest(destructiveChangesType)),
                    await cs.getPackageXml(4, destructiveChangesType)
                  );
                );
              }
```